### DISCUSSION TITLE
Undo in selection not working as expected after beacon edit

### DISCUSSION THREAD

**Discussion by fedreg-bn** [meow-edit/meow#102] [General]
Quick question on using `meow-undo-in-selection` after a beacon edit...

So say I have a tiny function...
```clj
(defn test-fn1 []
  (println "hello")
  (println "world"))
```
and I use a beacon edit to modify the two `println`s to...
```clj
(defn test-fn1 []
  (prn "hello")
  (prn "world"))
```
I then go off and edit some other code and when I try to do a `meow-undo-in-selection` by selecting the entire `test-fn1`, only one of the `prn` lines is undone.. the one that I did the actual insert command on.

Is there some other step I'd need to get the undo to work for all the lines I edited with beacon?

(I hope I explained that clearly enough!!)

Am LOVING meow!!  Thanks for all the awesome work

**Comment by DogLooksGood** [comment]
I can reproduce. 

I think the problem here is: meow will collapse operations into a sigle undo, but that's only for macro application.

Let's say you have three `println` here, you change the first to `prn`, and the rest two will be modified by kmacros automatically.
Now to undo all modifications you have to undo twice. The first is for macro application(the 2nd and 3rd `prn`), the second is for the real operations(the 1st `prn`).

And undo in region can be used for only once, since it will cancel the region.

I don't have a solution yet, we can not collape everything into one undo boundary.

**Reply by fedreg-bn** [reply]
Got it, thanks for the explanation!!

Nothing important... just something I noticed and thought I'd ask about.

Thanks again!
